File: leetcode/count-number-of-texts.py
class Solution:
    def countTexts(self, s: str) -> int:
        p = 1000000007
        f = [1] + [0] * len(s)
        for i in range(len(s)):
            f[i + 1] = f[i]
            if i > 0 and s[i] == s[i - 1]:
                f[i + 1] = (f[i + 1] + f[i - 1]) % p
                if i > 1 and s[i] == s[i - 2]:
                    f[i + 1] = (f[i + 1] + f[i - 2]) % p
                    if i > 2 and s[i] == s[i - 3] and s[i] in ['7', '9']:
                        f[i + 1] = (f[i + 1] + f[i - 3]) % p
        return f[-1]

File: leetcode/count-number-of-texts.cpp
class Solution {
public:
    int countTexts(string s) {
        const int p = 1000000007;
        int n = s.size();
        vector<int> f(n + 1);
        f[0] = 1;
        for (int i = 1; i <= n; i++)
        {
            f[i] = f[i - 1];
            if (i >= 2 && s[i - 1] == s[i - 2])
            {
                f[i] = (f[i] + f[i - 2]) % p;
            }
            if (i >= 3 && s[i - 1] == s[i - 2] && s[i - 2] == s[i - 3])
            {
                f[i] = (f[i] + f[i - 3]) % p;
            }
            if (i >= 4 && s[i - 1] == s[i - 2] && s[i - 2] == s[i - 3] && s[i - 3] == s[i - 4] && (s[i - 1] == '7' || s[i - 1] == '9'))
            {
                f[i] = (f[i] + f[i - 4]) % p;
            }
        }
        return f[n];
        
    }
};